server/jetty/build.gradle

```diff
@@ -69,3 +69,7 @@ applicationDefaultJvmArgs = [
 ] + extraJvmArgs
 
 apply plugin: 'io.deephaven.java-open-nio'
+
+tasks.withType(JavaExec) {
+    jvmArgs applicationDefaultJvmArgs
+}
```

Comment:
As formulated, this likely overwrites what we've set in java-open-nio.

I *think* `jvmArgs += applicationDefaultJvmArgs` would be better?

Reply:
I'll confirm this when i'm back, but applicationDefaultJvmArgs was already itself picking up that addopens, so I don't think it should need further processing?

```
> Task :server-jetty:JettyMain.main()
Caching disabled for task ':server-jetty:JettyMain.main()' because:
  Build cache is disabled
Task ':server-jetty:JettyMain.main()' is not up-to-date because:
  Task has not declared any outputs despite executing actions.
Starting process 'command '/usr/lib/jvm/java-11-openjdk/bin/java''. Working directory: /home/colin/workspace/illumon/core.new Command: /usr/lib/jvm/java-11-openjdk/bin/java --add-opens java.base/java.nio=ALL-UNNAMED -server -XX:+UseG1GC -XX:MaxGCPauseMillis=100 -XX:+UseStringDeduplication -XshowSettings:vm -XX:InitialRAMPercentage=25.0 -XX:MinRAMPercentage=70.0 -XX:MaxRAMPercentage=80.0 -Dfile.encoding=UTF-8 -Duser.country=US -Duser.language=en -Duser...
```

ACtually it looks like there is some other interaction going on here, like the :run itself is a JavaExec with defaults that come from applicationDefaultJvmArgs _appended_ to the task's own args, so they get doubled:
```
> Task :server-jetty:run
Caching disabled for task ':server-jetty:run' because:
  Build cache is disabled
Task ':server-jetty:run' is not up-to-date because:
  Task has not declared any outputs despite executing actions.
Starting process 'command '/usr/lib/jvm/java-11-openjdk/bin/java''. Working directory: /home/colin/workspace/illumon/core.new/server/jetty Command: /usr/lib/jvm/java-11-openjdk/bin/java -Ddeephaven.console.type=groovy -server -XX:+UseG1GC -XX:MaxGCPauseMillis=100 -XX:+UseStringDeduplication -XshowSettings:vm -XX:InitialRAMPercentage=25.0 -XX:MinRAMPercentage=70.0 -XX:MaxRAMPercentage=80.0 --add-opens java.base/java.nio=ALL-UNNAMED -server -XX:+UseG1GC -XX:MaxGCPauseMillis=100 -XX:+UseStringDeduplication -XshowSettings:vm -XX:InitialRAMPercentage=25.0 -XX:MinRAMPercentage=70.0 -XX:MaxRAMPercentage=80.0 -Dfile.encoding=UTF-8 -Duser.country=US -Duser.language=en -Duser...
```

Reply:
Some notes on this:

 * the `jvmArgs(...)` method strictly appends. To replace, use setJvmArgs(...), which clears the _extra_ jvm args and then reassigns, or setAllJvmArgs(...) which clears other properties as well. See org.gradle.process.internal.JvmOptions
 * It also parses the incoming flags, and passes some off to other fields/methods, like assertionsEnabled, systemProperty(), etc.
 * This means that interacting with "jvmArgs" in a JavaExec closure behaves differently based on using it like a method or a field. I don't love groovy. However, it does mean that this code is the correct way to assign these missing properties _for IJ run configs_, if I had written `jvmArgs = applicationDefaultJvmArgs`, it would have done as you described, mostly.
 * Repeating this for clarity: Note that the setJvmArgs method does _not_ unassign Xmx or system props, but you can at least re-assign them. Doesn't apply here, but it could bite one day.

Okay, knowing that, we want to set three options for three things:
 1. script for dists
 2. JavaExec for the `:run` task
 3. other arbitrary JavaExecs

A `tasks.withType(JavaExec)` lets us set 2 and 3. A `applicationDefaultJvmArgs` sets 1 and 2. `tasks.withType(CreateStartScripts)` only sets 1. So applying tasks.withType for both JavaExec and CreateStartScripts seems like the simplest option, and omit applicationDefaultJvmArgs.

Reply:
Tested with running each of these three ways, and invoking `jps -v | grep JettyMain` to check for missing/dup args.

Reply:
Good to know, thanks for the investigation.